src/network/netdev/wireguard.c

```diff
@@ -282,37 +282,42 @@ static int wireguard_resolve_handler(sd_resolve_query *q,
         return 0;
 }
 
-static void resolve_endpoints(NetDev *netdev) {
-        int r = 0;
-        Wireguard *w;
-        WireguardEndpoint *endpoint;
+static int resolve_endpoints(NetDev *netdev) {
         static const struct addrinfo hints = {
                 .ai_family = AF_UNSPEC,
                 .ai_socktype = SOCK_DGRAM,
                 .ai_protocol = IPPROTO_UDP
         };
+        WireguardEndpoint *endpoint;
+        Wireguard *w;
+        int k, r = 0;
 
         assert(netdev);
         w = WIREGUARD(netdev);
         assert(w);
 
         LIST_FOREACH(endpoints, endpoint, w->unresolved_endpoints) {
-                r = sd_resolve_getaddrinfo(netdev->manager->resolve,
+                k = sd_resolve_getaddrinfo(netdev->manager->resolve,
                                            &w->resolve_query,
                                            endpoint->host,
                                            endpoint->port,
                                            &hints,
                                            wireguard_resolve_handler,
                                            endpoint);
 
-                if (r == -ENOBUFS)
-                        break;
+                if (k == -ENOBUFS)
+                        return k;
```

Comment:
There is a limit of concurrent allowed queries. The break is needed if an endpoint exceed this limit (i.e. an
endpoint can have 10000 of peers). Therefore please revert this last commit.

Reply:
Ah, I see. Thanks. Dropped.